### DISCUSSION TITLE
How do I share components between react and react-native?

### DISCUSSION THREAD

**Discussion by robert-bitguild** [necolas/react-native-web#2397] [Q&A] [answered]
I want to share UI components between react and react native. So I create a UI components project by react-native-web and I want to compile the project as a UI library which used in both react and react-native projects.  How can I achieve the target?

Thanks.

**Comment by yuxizhe** [comment]
Sure you can. React-native-web make it happen. I already build a tiny universal ui components library with styled-system https://snowbox.js.org/
You can learn from opensouce ui library nativebase  https://nativebase.io/   or  tamagui https://tamagui.dev/  they both use react-native-web

**Comment by necolas** [comment, marked as answer]
react-native-web is a web implementation of most of the react-native API. The [documentation](https://necolas.github.io/react-native-web/docs/setup/) describes how it should be used as a package alias for `react-native` when bundling apps for the web. To create a cross-platform UI library, you can use the `react-native` package and API in the components, and then use `react-native-web` package aliasing to bundle the UI library for web.
